tests/generic_config_updater/test_eth_interface.py

```diff
@@ -0,0 +1,148 @@
+import ipaddress
+import logging
+import pytest
+import re
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(autouse=True)
+def ensure_dut_readiness(duthost):
+    """
+    Setup/teardown fixture for each ipv6 test
+    rollback to check if it goes back to starting config
+
+    Args:
+        duthost: DUT host object under test
+    """
+
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+
+def check_interface_status(duthost, field):
+    """
+    Returns current status for Ethernet0 of specified field
+
+    Args:
+        duthost: DUT host object under test
+        field: interface field under test
+    """
+
+    cmds = "show interface status Ethernet0"
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'])
+    status_data = output["stdout_lines"]
+    
+    field_index = status_data[0].split().index(field)
+    status = status_data[2].split()[field_index]
+    return status
+    
+
+@pytest.mark.parametrize("index, is_valid", [
+    ("33", True),
+    ("abc1", False)
+])
+def test_update_valid_invalid_index(duthost, ensure_dut_readiness, index, is_valid):
```

Comment:
>test_update_valid_invalid_index [](http://example.com/codeflow?start=4&length=31)

Other useful test cases:
1. replace/delete lanes, which will be banned by Yang model
2. replace mtu
3. toggle pfc_asym on/off: need to investigate how to valid
4. replace fec

Reply:
Thank you, added these test cases

Reply:
Hmm, why `replace lanes` should fail by Yang model? 

Yang model only have the lanes `Required`, but the lanes to be of different values is OK.

`lanes` field is not only `required` but also `create-only`. GCU will first delete the whole `PORT/Ethernet0` object and add it back with the new lanes.